### PR TITLE
US283 Readme Upgrade

### PR DESCRIPTION
**Pull request by aneuhold**
### Overview

This PR implements US 283, the readme and documentation upgrade. Here is a [quick link to how the main readme looks](https://github.com/halomod/TheHaloMod-SPA/tree/US283-ReadmeUpgrade). Existing popular readmes were used as inspiration to make this one.

The main `README.md` was updated as well as the `server` and `client` `README.md` files. They work together to explain the project. 

Badges were added too, which automatically update. They were pretty easy to add with [shields.io](https://shields.io/).

Documentation was added in a few places and updated in others. This isn't exhaustive of course. Some of the things in the project I do not fully understand, so it might be better for people to add documentation as they go for those parts that they remember that doesn't already have something there. Some documentation was also not added because the components are being changed currently in other branches.

Our names with links to our GitHub profiles was added to the About page as well 😊.

### Includes:
- LICENSE 
- README.md 
- client/README.md 
- client/package-lock.json 
- client/package.json 
- client/src/components/Download.vue 
- client/src/components/Navbar.vue 
- client/src/main.js 
- client/src/router/index.js 
- client/src/utils/Store.js 
- client/src/utils/plot.js 
- client/src/views/About.vue 
- client/src/views/Forms.vue 
- client/src/views/Home.vue 
- client/yarn.lock 
- server/README.md 
- server/config.py 
- server/halomod_app/__init__.py 
- server/halomod_app/utils.py 
- server/tests/conftest.py 
- server/tests/test_halomod.py 

### Developer Checklist:
- [x] The code follows the Quality Policy file on Google Drive
- [N/A] My code has been developer-tested and includes unit tests
- [x] I have considered proper use of exceptions
- [x] I have eliminated IDE warnings
- [x] I have included tasks associated with this pull request in the Sprint Progress Sheet

### Notes:
- A license was added that was the same as the license on `TheHaloMod`. This is currently saying "not specified" in the badge because it looks for the license on the main branch. So once the license is pulled into `main`, it will reflect that.
- Some unused packages were removed.

### Refs:
- [US 283](https://the-halo-mod.atlassian.net/jira/software/projects/HM/boards/1?selectedIssue=HM-283)

### CODE REVIEW

File: client/README.md [hunk 1]
```diff
@@ -1,29 +1,51 @@
...
+1. Run `npm install`. (If you don't have npm installed, [see here for instructions](https://github.com/nvm-sh/nvm#installing-and-updating)).
+1. Run `npm run serve`
+1. The client should be available on `localhost:8080`
+
+### How to build and run the production version of the client locally
+
+1. Make sure you are in the `TheHaloMod-SPA/client` directory in your shell of choice
+1. Run `npm install` if that hasn't already been done. (If you don't have npm installed, [see here for instructions](https://github.com/nvm-sh/nvm#installing-and-updating)).
+1. Run `npm run build` and wait for it to finish
+1. Run `serve dist -p 5111`. Note that `-p 5111` sets the port to `5111` because by default the port is set to `5000` which the server is running on. So `5111` can be any port that you choose that isn't already taken. Also if `serve` isn't installed, [see here for instructions](https://www.npmjs.com/package/serve).
```
Comment (Tengelma) [inline client/README.md:27]: Should probably actually number these instead of just all ones lol

File: server/README.md [hunk 2]
```diff
@@ -1,13 +1,41 @@
...
+
+1. Make sure you are in the `server` directory in your shell of choice
+1. Run `. ./run.sh --install`. This will install all of the dependencies in [`requirements.txt`](requirements.txt) and enter into a [virtual environment](https://docs.python.org/3/tutorial/venv.html).
+
+### How to start the server locally
+
+1. Open a second shell window (secondary to the shell that has a working directory of `server`. See the first step above.)
+1. In the second shell window, start up a [Redis DB](https://redis.io/) by running `redis-server`, which may need to be installed first. If this isn't installed yet, [see here on how to install it](https://redis.io/topics/quickstart).
+1. In the first shell window, run `. ./run.sh --dev`
+1. The server should now be located at `localhost:5000`
```
Comment (Tengelma) [inline server/README.md:24]: Same as other comment.

**Review by Tengelma — APPROVED** [review]
Approving because the changes are minor and low priority but if you want to throw another commit in to fix them you will get an extra extra approval. Screenshots of the rendered markdown would also be nice just to make it a little easier for the next reviewer.

This is so much better than what we had & the fact that you updated file / function documentation too is much appreciated. Goes quite a ways in making the project more understandable and that was the main goal of the User Story so :thumbsup:

**Review by vperuzzi — APPROVED** [review]
Sorry it too me so long to approve this. Anton, this documentation is extensive and excellent. The function comments and documentation is wonderful. Thank you for taking the time to do this. Approving.

File: client/README.md (hunk 1 above)
Comment (aneuhold) [reply client/README.md:27]: @Tengelma The reason it is done that way is so if a step needs to be moved around they don't all need to be re-numbered again. This way it is left to the renderer to do it's thang. With that said, do you still want it to be numbered in the code?

Comment (Tengelma) [reply client/README.md:27]: Ah, didn't realize renderer just handled it. We're all good then. Nice work!

Comment (aneuhold) [reply client/README.md:27]: Rock on. Thanks for taking a look 😄

**Pull request merged**